applications/GeoMechanicsApplication/custom_constitutive/truss_backbone_constitutive_law.cpp

```diff
@@ -0,0 +1,223 @@
+    }
+    mPreviousAxialStrain = axial_strain;
+}
+
+int TrussBackboneConstitutiveLaw::Check(const Properties&   rMaterialProperties,
+                                        const GeometryType& rElementGeometry,
+                                        const ProcessInfo&  rCurrentProcessInfo) const
+{
+    if (const auto exit_code = ConstitutiveLaw::Check(rMaterialProperties, rElementGeometry, rCurrentProcessInfo);
+        exit_code != 0)
+        return exit_code;
+
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(YOUNG_MODULUS)) << "No YOUNGS_MODULUS found" << std::endl;
+
+    CheckStressStrainDiagram(rMaterialProperties);
+
+    return 0;
+}
+
+double TrussBackboneConstitutiveLaw::BackboneStress(double Strain) const
+{
+    return mStressStrainTable.GetValue(Strain);
+}
+
+double TrussBackboneConstitutiveLaw::BackboneStiffness([[maybe_unused]] double Strain) const
+{
+    return mStressStrainTable.GetDerivative(Strain);
+}
+
+double TrussBackboneConstitutiveLaw::CalculateUnReLoadAmplitude(double YoungsModulus) const
+{
+    return BackboneStress(mAccumulatedStrain) / YoungsModulus;
+}
+
+bool TrussBackboneConstitutiveLaw::IsWithinUnReLoading(double Strain, double YoungsModulus) const
+{
+    return std::abs(Strain - mUnReLoadCenter) < CalculateUnReLoadAmplitude(YoungsModulus);
+}
+
+bool TrussBackboneConstitutiveLaw::RequiresInitializeMaterialResponse() { return true; }
+
+SizeType TrussBackboneConstitutiveLaw::GetStrainSize() const { return 1; }
+
+void TrussBackboneConstitutiveLaw::save(Serializer& rSerializer) const
+{
+    KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, BaseType)
+    rSerializer.save("AccumulatedStrain", mAccumulatedStrain);
+    rSerializer.save("PreviousAxialStrain", mPreviousAxialStrain);
+    rSerializer.save("UnReload", mUnReLoadCenter);
+}
+
+void TrussBackboneConstitutiveLaw::load(Serializer& rSerializer)
+{
+    KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, BaseType)
+    rSerializer.load("AccumulatedStrain", mAccumulatedStrain);
+    rSerializer.load("PreviousAxialStrain", mPreviousAxialStrain);
+    rSerializer.load("UnReload", mUnReLoadCenter);
+}
+
+void TrussBackboneConstitutiveLaw::InitializeMaterial(const Properties& rMaterialProperties,
+                                                      const ConstitutiveLaw::GeometryType& rElementGeometry,
+                                                      const Vector& rShapeFunctionsValues)
+{
+    ConstitutiveLaw::InitializeMaterial(rMaterialProperties, rElementGeometry, rShapeFunctionsValues);
+
+    const auto& r_strains_of_piecewise_linear_law = rMaterialProperties[STRAINS_OF_PIECEWISE_LINEAR_LAW];
+    const auto& r_stresses_of_piecewise_linear_law = rMaterialProperties[STRESSES_OF_PIECEWISE_LINEAR_LAW];
+    for (auto i = std::size_t{0}; i < r_strains_of_piecewise_linear_law.size(); ++i) {
+        mStressStrainTable.PushBack(r_strains_of_piecewise_linear_law[i],
+                                    r_stresses_of_piecewise_linear_law[i]);
+    }
```

Comment:
@rfaasse : What do you think? Should we empty the table prior to filling it (to allow for this initialization to be executed more than once)?